keepercommander/importer/imp_exp.py

```diff
@@ -678,211 +1064,269 @@ def prepare_folder_add(params, folders, records):
     return folder_add
 
 
-def tokenize_partial_import_record(record):   # type: (ImportRecord) -> [str]
+def parepare_record_audit(params, uids, enterprise_public_key):  # type: (KeeperParams, Iterator[str], any) -> Iterator[record_proto.RecordAddAuditData]
+    for uid in uids:
+        if uid in params.record_cache:
+            keeper_record = params.record_cache[uid]
+            import_record = convert_keeper_record(keeper_record)
+            if import_record:
+                audit_data = {
+                    'title': import_record.title,
+                    'record_type': import_record.type
+                }
+                if import_record.login_url:
+                    audit_data['url'] = import_record.login_url
+                record_audit_rq = record_proto.RecordAddAuditData()
+                record_audit_rq.record_uid = utils.base64_url_decode(uid)
+                record_audit_rq.revision = 0
+                record_audit_rq.data = crypto.encrypt_ec(json.dumps(audit_data).encode('utf-8'), enterprise_public_key)
+                yield record_audit_rq
+
+
+RECORD_FIELD_TYPES = {'login', 'password', 'url'}
+IGNORABLE_FIELD_TYPES = {'text', 'fileRef', 'cardRef', 'oneTimeCode'}
+
+
+def value_to_token(value): # type: (any) -> str
+    if not value:
+        return ''
+    if type(value) == str:
+        return value
+    if type(value) == list:
+        return ','.join((value_to_token(x) for x in value))
+    if type(value) == dict:
+        pairs = [x for x in value.items()]
```

Comment:
FWIW, this could probably be simplified a little to pairs = list(value.items())

Reply:
Python can compete with Perl on `There's more than one way to do it / TMTOWTDI`